packages/core/src/domain/internalMonitoring.ts

```diff
@@ -101,26 +101,41 @@ export function monitored<T extends (...params: any[]) => unknown>(
   descriptor: TypedPropertyDescriptor<T>
 ) {
   const originalMethod = descriptor.value!
-  descriptor.value = function (this: any, ...args: Parameters<T>) {
+  descriptor.value = function (this: any, ...args: utils.Parameters<T>) {
     const decorated = monitoringConfiguration.batch ? monitor(originalMethod) : originalMethod
     return decorated.apply(this, args) as ReturnType<T>
   } as T
 }
 
 export function monitor<T extends (...args: any[]) => any>(fn: T): T {
   return (function (this: any) {
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-return
+    return callMonitored(fn, this, (arguments as unknown) as utils.Parameters<T>)
+  } as unknown) as T // consider output type has input type
+}
+
+export function callMonitored<T extends (...args: any[]) => any>(
+  fn: T,
+  context: utils.ThisParameterType<T>,
+  args: utils.Parameters<T>
+): ReturnType<T> | undefined
+export function callMonitored<T extends (this: void) => any>(fn: T): ReturnType<T> | undefined
+export function callMonitored<T extends (...args: any[]) => any>(
+  fn: T,
+  context?: any,
+  args?: any
+): ReturnType<T> | undefined {
+  try {
```

Comment:
> This commit introduces callMonitored function to allow a more granular
monitoring of our code.

what was wrong with:
```ts
monitor(fn)()
```

Reply:
It creates a useless intermediate function.

Reply:
hum, that quite some ts boilerplate to avoid intermediate functions.
I'm also a bit bothered by the naming but I have nothing better to propose.

unless you have a better idea let's go with that.